adapters/wazirx-nft-Xauction-bsc/index.js

```diff
@@ -0,0 +1,176 @@
+            address: null,
+            price: 0,
+        };
+
+        const tokenId = event.returnValues._tokenId;
+
+        const transaction = await this.sdk.getTransactionReceipt(event.transactionHash);
+
+        const auctionContract = this.sdk.web3.eth.abi.encodeParameter("address", this.contract);
+        const logs = transaction.logs.filter(
+            tx => tx.topics[0] === TRANSFER_TOPIC && tx.topics[1].toLowerCase() === auctionContract,
+        );
+
+        const bnbusd = await this.getPrice(block.timestamp, this.currencyCode["2"].token_address);
+
+
+        if (logs.length) {
+            // Look for highest amount transfer
+            const price = logs.reduce((sum, item, index) => {
+                const decodedPrice = new BigNumber(item.data).dividedBy(10 ** 8).toNumber();
+                const address = this.sdk.web3.eth.abi.decodeParameter("address", item.topics[2]);
+                sum += decodedPrice;
+
+                if (index === 0) {
+                    seller = {
+                        address: address,
+                        price: decodedPrice,
+                    };
+                } else if (decodedPrice > seller.price) {
+                    seller = {
+                        address: address,
+                        price: decodedPrice,
+                    };
+                }
+                return sum;
+            }, 0);
+            po = await this.getPrice(block.timestamp, this.currencyCode["1"].token_address);
+            nativePrice = new BigNumber(price).times(po.price).dividedBy(bnbusd.price);
+        } else {
+            let events = [];
+            let _event = {};
+            let blockRange = event.blockNumber;
+            // Look for latest BidPlaced event by bid winner to find price
+            while (!events.length) {
+                events = await this.sdk.getPastEvents("BidPlaced", {
+                    filter: { _tokenId: tokenId, _from: event.returnValues._bidWinner, _currency: "2" },
+                    fromBlock: blockRange - 5000,
+                    toBlock: blockRange,
+                });
+                blockRange -= 5000;
+            }
+
+            _event = events[events.length - 1];
+            nativePrice = new BigNumber(_event.returnValues._amount).dividedBy(10 ** 18);
+            po = bnbusd;
+            seller = {
+                address: event.returnValues._settledBy,
+            };
+        }
+
+        const entity = {
+            provider_name: this.name,
+            provider_contract: this.contract,
+            protocol: AVAILABLE_PROTOCOLS.BSC,
+            nft_contract: this.nftContract,
+            nft_id: tokenId,
+            token: this.token,
+            token_symbol: this.token_symbol,
+            amount: 1,
+            price: nativePrice.toNumber() || 1,
+            price_usd: nativePrice.multipliedBy(po.price).toNumber(),
+            seller: seller.address,
+            buyer: event.returnValues._bidWinner,
```

Comment:
Please lowercase `seller` and `buyer` properties. All addresses has to be lowercased before returning sale object.